packages/grid/_modules_/grid/components/containers/GridRoot.tsx

```diff
@@ -9,26 +14,36 @@ import { GridApiContext } from '../GridApiContext';
 export type GridRootProps = React.HTMLAttributes<HTMLDivElement>;
 
 export const GridRoot = React.forwardRef<HTMLDivElement, GridRootProps>(function GridRoot(
-  props,
+  { children },
   ref,
 ) {
-  const { className, ...other } = props;
   const classes = useStyles();
-  const apiRef = React.useContext(GridApiContext);
+  const apiRef = React.useContext(GridApiContext)!;
+  const props = React.useContext(GridPropsContext)!;
+  const { className } = props;
   const visibleColumnsLength = useGridSelector(apiRef, visibleGridColumnsLengthSelector);
   const [gridState] = useGridState(apiRef!);
+  const options = useGridSelector(apiRef, optionsSelector);
+  const rootContainerRef: GridRootContainerRef = React.useRef<HTMLDivElement>(null);
+  const handleRef = useForkRef(rootContainerRef, ref);
+  apiRef.current.rootElementRef = rootContainerRef;
 
   return (
-    <div
-      ref={ref}
-      className={clsx(classes.root, className, {
-        'MuiDataGrid-autoHeight': gridState.options.autoHeight,
-      })}
-      role="grid"
-      aria-colcount={visibleColumnsLength}
-      aria-rowcount={gridState.rows.totalRowCount}
-      aria-multiselectable={!gridState.options.disableMultipleSelection}
-      {...other}
-    />
+    <NoSsr>
+      <div
+        ref={handleRef}
+        className={clsx(classes.root, options.classes?.root, className, {
+          'MuiDataGrid-autoHeight': gridState.options.autoHeight,
+        })}
+        role="grid"
+        aria-colcount={visibleColumnsLength}
+        aria-rowcount={gridState.rows.totalRowCount}
+        aria-multiselectable={!gridState.options.disableMultipleSelection}
+        aria-label={props['aria-label']}
+        aria-labelledby={props['aria-labelledby']}
+      >
+        {children}
+      </div>
```

Comment:
Match the conformance tests

```suggestion
      {...other}
    />
```

Reply:
GridRoot cannot have other props at this stage, see GridComponent

Reply:
I think that it depends on the case:

- If GridRoot is a private component, the main value I see in using {...other} is to simplify the source of it.
- If GridRoot s a public component, then we should spread all the props on the root as soon as we can (in this case, it seems that we can)

Reply:
Done. But we can't have other props at this stage, so IMO we could have done this refactoring when we make gridRoot public, as it brings a bit of complexity for something that can't work atm. So the next guy would wonder why I've done that.

Reply:
>  But we can't have other props at this stage

So you are saying it's public? Then, why should we prevent a developer to add a onClick listener for instance?